x/ccv/provider/keeper/throttle.go

```diff
@@ -19,54 +19,56 @@ import (
 func (k Keeper) HandleThrottleQueues(ctx sdktypes.Context) {
 
 	meter := k.GetSlashMeter(ctx)
-
-	// Don't start iterating if meter is negative in value
+	// Return if meter is negative in value
 	if meter.IsNegative() {
 		return
 	}
 
-	handledGlobalEntries := []providertypes.GlobalSlashEntry{}
-
-	// Iterate through ordered (by received time) global slash entries from any consumer chain
-	k.IterateGlobalSlashEntries(ctx, func(entry providertypes.GlobalSlashEntry) (stop bool) {
-
-		// Obtain validator from the provider's consensus address.
-		// Note: if validator is not found or unbonded, this will be handled appropriately in HandleSlashPacket
-		val, found := k.stakingKeeper.GetValidatorByConsAddr(ctx, entry.ProviderValConsAddr)
-
-		// Obtain the validator power relevant to the slash packet that's about to be handled
-		// (this power will be removed via jailing or tombstoning)
-		var valPower int64
-		if !found || val.IsJailed() {
-			// If validator is not found, or found but jailed, it's power is 0. This path is explicitly defined since the
-			// staking keeper's LastValidatorPower values are not updated till the staking keeper's endblocker.
-			valPower = 0
-		} else {
-			valPower = k.stakingKeeper.GetLastValidatorPower(ctx, val.GetOperator())
-		}
+	// Obtain all global slash entries, where only some of them may be handled in this method,
+	// depending on the value of the slash meter.
+	allEntries := k.GetAllGlobalSlashEntries(ctx)
```

Comment:
My concern here is that the size of this slice depends on the number of slash packets received from consumer chains. This is related to https://github.com/cosmos/interchain-security/issues/594. A solution here would be to have a filter on the getter that only returns the entries the throttling mechanism can handled.